vumi/transports/wechat/wechat.py

```diff
@@ -114,23 +115,30 @@ def render_GET(self, request):
         request.setResponseCode(http.BAD_REQUEST)
         return ''
 
-    def render_POST(self, request):
+    @inlineCallbacks
+    def validate_request(self, request):
         if not (is_verifiable(request)
                 and verify(self.config.auth_token, request)):
-            request.setResponseCode(http.BAD_REQUEST)
-            return ''
+            raise WeChatException('Bad request for incoming message')
+        elif is_verifiable(request):
```

Comment:
I don't think we need to check `is_verifiable` again here. If we don't raise the exception, we should always call `add_status_good_req`?